framework/Source/Kirakira.swift

```diff
@@ -10,9 +10,87 @@ import Foundation
 
 public class Kirakira: OperationGroup {
 
-    public enum ColorMode: Int {
+    public enum ColorMode: Int, Codable {
         case white = 0
         case random = 1
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.singleValueContainer()
+            let rawValue = try container.decode(Int.self)
+            self = ColorMode(rawValue: rawValue) ?? .random
+        }
+
+        public func encode(to encoder: Encoder) throws {
+            var container = encoder.singleValueContainer()
+            try container.encode(rawValue)
+        }
+    }
+
+    public struct Parameters: Codable {
```

Comment:
Hi @yyjim, I want to discuss the following 2 things with you:
1. Your example code was confirmed Codable, but will you have concerns if I only make it to confirm Decodable?
  - The Kirakira effect here is positioning as just a filter for PicCollage, which means we won't encode it into our collage structure directly
  - Therefore, we only have to decode it from JSON from the Explore team.
  
2. Do you want me to implement the decode with the JSON format from Explore? I notice they have a specific format instead of just using property name as the key and property value as the value
```JSON
    {
        "identifier" : "15D817B1-CA3B-42C3-AA92-946A0CCFA102",
        "name" : "Kirakira - Flashlight",
        "visualEffects" : [
            {
                "parameters" : {
                    "equalMinHue" : {
                        "type" : "float",
                        "value" : 0.75
                    },
                    "equalMaxHue" : {
                        "type" : "float",
                        "value" : 0.08
                    },
                    "equalSaturation" : {
                        "type" : "float",
                        "value" : 0.15
                    },
                    "equalBrightness" : {
                        "type" : "float",
                        "value" : 2
                    },
                    "speed" : {
                        "type" : "float",
                        "value" : 7.5
                    },
                    "rayCount" : {
                        "type" : "int",
                        "value" : 4
                    },
                    "rayLength" : {
                        "type" : "float",
                        "value" : 0.12
                    },
                    "sparkleExposure" : {
                        "value" : 0.70,
                        "type" : "float"
                    },
                    "blur" : {
                        "value" : 3,
                        "type" : "int"
                    },
                    "colorMode" : {
                        "value" : 0,
                        "type" : "int"
                    },
                    "saturation" : {
                        "type" : "float",
                        "value" : 0.3
                    },
                    "centerSaturation" : {
                        "type" : "float",
                        "value" : 0.3
                    },
                    "minHue" : {
                        "value" : 0,
                        "type" : "float"
                    },
                    "maxHue" : {
                        "value" : 1,
                        "type" : "float"
                    },
                    "noiseInfluence" : {
                        "type" : "float",
                        "value" : 1
                    },
                    "increasingRate" : {
                        "value" : 0,
                        "type" : "float"
                    },
                    "startAngle" : {
                        "type" : "int",
                        "value" : 26
                    },
                    "sparkleAmount" : {
                        "type" : "float",
                        "value" : 0
                    },
                    "frameRate" : {
                        "type" : "int",
                        "value" : 60
                    }
                },
                "effectClassName" : "CBVisualEffectBuiltins.KirakiraEffect"
            }
        ]
    }
```

Reply:
Only need `Decodable` for parameters. no need to decode other properties.